Added Steps for Reporting issue on Github (#5934)
jenkins-infra/jenkins.io: content/participate/report-issue.adoc

== How-To Report an Issue

> Jenkins JIRA is an issue tracker not a support site
> If you need assistance or have general questions, visit
> us link:/chat/[in chat], email one of the link:/mailing-lists[mailing lists], 
> or post on the https://community.jenkins.io/[community forum].

Reviewer: ```suggestion
Jenkins JIRA is an issue tracker, not a support site.
If you need assistance or have general questions, visit us link:/chat/[in chat], email one of the link:/mailing-lists[mailing lists], or post on the https://community.jenkins.io/[community forum].
```
update sentence structure and punctuation

[[Howtoreportanissue-Beforecreatinganissue]]
== Before creating an issue

> First, *identify the location of the issue*. Are you reporting an issue
> in Jenkins itself or one of its plugin. We document all the issues in 
> Jenkins on link:https://issues.jenkins.io/[JIRA] and some plugin's
> issues are on Github.
> 
> After that determine the what kind of issue it is. There are three main types

Reviewer: ```suggestion
First, *identify the location of the issue*.
Are you reporting an issue in Jenkins itself or one of its plugins? 
We document all the issues in Jenkins on link:https://issues.jenkins.io/[JIRA] and plugin issues are on Github.
If you are reporting an issue with the jenkins.io site, please create an issue in our link:https://github.com/jenkins-infra/jenkins.io/issues[GitHub issue tracker].

After that, determine what kind of issue it is. There are three main types:
```
update structure and syntax, add link for jenkins.io github repo

whose use is explained on this page:

* *Bug*: Something does not work as intended.

> * *Improvement*: Either current functionality should be improved or new 
> functionality should be added.
> * *Vulnerabilities*: link:https://www.jenkins.io/security/[Security related issues in Jenkins]

Reviewer: ```suggestion
* *Improvement*: Either a current functionality should be improved or new 
functionality should be added.
* *Vulnerabilities*: link:/security/reporting/[Security related issues in Jenkins]
```
update structure and link

> Other types (_Patch_, _Task_) are not used in JIRA. Neither will

Reviewer: ```suggestion
Other types (_Patch_, _Task_) are not used in JIRA.
Neither will be explained here.
If you want to propose a patch, create an issue using one of the 3 types above, and then create a link:https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request[pull request in GitHub].
```
update sentence structure for entire paragraph below

https://stackoverflow.com/tags/jenkins[Stack Overflow] may also be a
good place to search. Any results will often point you to earlier issue
reports if it's really a bug. You can then _watch them_ and _vote for

> them_ in JIRA or subscribe for notifications in Github Issues.
> Voting helps us determine the impact of bugs and prioritize them.
> _If you have doubts that the problem you're experiencing is the 
> exact same one as you found, prefer to file a new

Reviewer: ```suggestion
https://stackoverflow.com/tags/jenkins[Stack Overflow] may also be a
good place to search.
Any results will often point you to earlier issue reports if it's really a bug.
You can then _watch them_ and _vote for them_ in JIRA, or subscribe to notifications in Github Issues.
Voting helps us determine the impact of bugs and prioritize them.
If you have doubts that the problem you're experiencing is the exact same one as you found, and prefer to file a new issue, be sure to mention the one(s) you found in the description.
```
update for sentence structure and syntax

directly by clicking the plugin name in _Manage Jenkins_ » _Manage Plugins_ » _Installed_.
If you're still unable to resolve the problem and cannot find existing
issue reports or related fixes in newer versions of Jenkins or the

> plugins, file a *bug*.

Reviewer: ```suggestion
If you're still unable to resolve the problem, and cannot find existing issue reports or related fixes in newer versions of Jenkins or the plugins, file a *bug*.
```
update sentence structure

[[Howtoreportanissue-Beforerequestinganimprovementorfeature]]
=== Before requesting an improvement or feature

* *Check newer versions of Jenkins or the plugin* for this feature.
They may have the feature already!
* *Search JIRA or Github Issues for existing requests* for this feature.
You can then _watch them and vote for them_. Voting helps us determine 
the demand for requests and prioritize them, so it's in your interest 
to actually find these rather than reporting a duplicate.

If you cannot find anything, file an *improvement* or *new feature*

> request in link:https://issues.jenkins.io/[JIRA] issue tracker for the
> specific plugin. Open the plugin site for the plugin and click the 
> "Report an issue" link in the "Links" section on the right side of the
> plugin page.

Reviewer: ```suggestion
If you cannot find anything, file an *improvement* or *new feature* request in the issue tracker for the specific plugin.
Open the link:https://plugins.jenkins.io/[page] for the plugin and select the "Report an issue" link in the *Links* section on the right side of the plugin page.
```
updated syntax/sentence structure

directly by clicking the plugin name in _Manage Jenkins_ » _Manage Plugins_ » _Installed_.
If you're still unable to resolve the problem and cannot find existing
issue reports or related fixes in newer versions of Jenkins or the
If you're still unable to resolve the problem, and cannot find existing issue reports or related fixes in newer versions of Jenkins or the plugins, file a *bug*.

[[Howtoreportanissue-Beforerequestinganimprovementorfeature]]
=== Before requesting an improvement or feature

* *Check newer versions of Jenkins or the plugin* for this feature.
They may have the feature already!
* *Search JIRA or Github Issues for existing requests* for this feature.
You can then _watch and vote for them_.
Voting helps us determine the demand for requests and prioritize them, so it's in your interest to actually find these rather than creating a duplicate.

> If you cannot find anything, file an *improvement* or *new feature*

Reviewer: ```suggestion

```
remove the extra line here as it is unnecessary

[[Howtoreportanissue-Creatingtheissue]]
== Creating the issue

> If you are reporting an issue in Jenkins, then you'll have to report 
> it on link:https://issues.jenkins.io/[JIRA]. However *if the issue you're
> reporting is in one of Jenkins' plugin*. you'll have to follow these steps:
> 
> * Open the link:https://plugins.jenkins.io/[Plugin Site]
> * Search for the Plugin and open its page
> * Click the *Report an issue* link in the *Links* section on the right side of the
> plugin page.
> ** After clicking the link if you end up on JIRA, follow the instructions on link:/participate/report-issue/#reporting-issue-on-jira[Reporting Issue in JIRA]
> ** After clicking the link if you end up on Github, follow the instructions on link:/participate/report-issue/#reporting-issue-on-github[Reporting Issue on Github]
> 

Reviewer: ```suggestion
If you are reporting an issue in Jenkins, then you'll have to report it on link:https://issues.jenkins.io/[JIRA].
However *if the issue you're reporting is in one of Jenkins' plugins*, you'll have to follow these steps:

* Open the link:https://plugins.jenkins.io/[Plugin Site].
* Search for the Plugin and open its page.
* Click the *Report an issue* link in the *Links* section on the right side of the
plugin page.
** After selecting the link, if you end up on JIRA follow the instructions on link:/participate/report-issue/#reporting-issue-on-jira[Reporting an Issue in JIRA].
** After selecting the link, if you end up on Github follow the instructions on link:/participate/report-issue/#reporting-issue-on-github[Reporting an Issue on Github].

```
update structure and syntax

it fixed*. Try to be realistic: Most bugs should be Minor or Major
priority. If you found a workaround, reduce the priority, even if it's a
severe problem otherwise.
* Leave the *Assignee* field to automatic.

=== Reporting an issue on GitHub 
You need to have a Github account. 
link:https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account[Register an account] if you haven't already.

* If you are redirected to the Github issue tracker from the link:https://plugins.jenkins.io/[Plugin Site], you will see something like this:
+
[.boxshadow]
image:/images/participate/github-issues.png[alt="Github Issue Page",width=80%]

* Select *Get started* next to the type of issue you'd like to open.
+
[.boxshadow]

> image:/images/participate/get-started.png[alt="Get Started Button",width=80%]

Reviewer: ```suggestion
image:/images/participate/get-started.png[alt="Get Started Button",width=40%]
```
In this case, I would say that the get started button does not need to be at 80%, as that is much bigger than the other images on the page. I would scale it down to either 40% or lower, or remove the width altogether. Using the `make` command to have a live preview site and then adjusting the size as needed would allow you to ensure that the image is closer to the rest of the page/images.

* After that, fill in the details of the issue according to the Github issue template provided for each type of issue.
* Don't forget to subscribe to the issue so you'll get notifications regarding any updates on the issue.

> [[Howtoreportanissue-WhatinformationtoprovideforEnvironmentandDescription]]
> === What information to provide for Environment and Description

Reviewer: ```suggestion
[[Howtoreportanissue-WhatinformationtoprovideforEnvironmentandDescription]]

=== What information to provide for Environment and Description
```
there needs to be an empty line between the anchor and the header so that the formatting can properly render.

it fixed*. Try to be realistic: Most bugs should be Minor or Major
priority. If you found a workaround, reduce the priority, even if it's a
severe problem otherwise.
* Leave the *Assignee* field to automatic.

> === Reporting Issue on Github 
> You need to have an Github Account if don't have already. link:https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account[Register an account] if you haven't already.
> 
> * If you are redirected to Github Issue section from the link:https://plugins.jenkins.io/[Plugin Site]
> you will see something like this
> 
> image:/images/participate/github-issues.png[alt="Github Issue Page",width=80%]
> 
> * According to the type of issue you are reporting,click Get started next to the 
> type of issue you'd like to open.
> 
> image:/images/participate/get-started.png[alt="Get Started Button",width=80%]
> 
> * After that fill the details of the issue according to the Github issue
> template given for each type of issue.
> * Don't Forget to subscribe the issue so you'll get notifications regarding the
> issue you created.
> 

Reviewer: ```suggestion
=== Reporting an issue on GitHub 
You need to have a Github account. 
link:https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account[Register an account] if you haven't already.

* If you are redirected to the Github issue tracker from the link:https://plugins.jenkins.io/[Plugin Site], you will see something like this:
+
image:/images/participate/github-issues.png[alt="Github Issue Page",width=80%]

* Select *Get started* next to the type of issue you'd like to open.
+
image:/images/participate/get-started.png[alt="Get Started Button",width=80%]

* After that, fill in the details of the issue according to the Github issue template provided for each type of issue.
* Don't forget to subscribe to the issue so you'll get notifications regarding any updates on the issue.
```
update to sentence structure and formatting so that images are inline with text
